Source/WPEFramework/CMakeLists.txt

```diff
@@ -69,7 +69,12 @@ target_link_libraries(${TARGET}
           ${NAMESPACE}COMProcess::${NAMESPACE}COMProcess  # For COM executable define
           Threads::Threads
         )
-        
+
+if(WARNING_REPORTING)
+    target_sources(${TARGET} PRIVATE WarningReportingCategories.cpp)
+    target_compile_definitions(${TARGET} PUBLIC __CORE_WARNING_REPORTING__)
```

Comment:
This looks a bit redundant. This flag should come from the WarningReporting CMake file and not from here as they should be completely independent.  
Also if the flag is set, I was expecting this to be the target_link_libraries(${TARGET}
        PRIVATE {NAMESPACE}WarningReporting::{NAMESPACE}WarningReporting)

@bramoosterhuis how can we make sure that the preprocessor statement (target_compile_definitions) is coming from the WarningReporting Cmake here: https://github.com/rdkcentral/Thunder/blob/master/Source/extensions/warningreporting/CMakeLists.txt#L32 ???